src/main/java/org/disq_bio/disq/HtsjdkVariantsRddStorage.java

```diff
@@ -0,0 +1,191 @@
+package org.disq_bio.disq;
+
+import htsjdk.samtools.util.Locatable;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFHeader;
+import java.io.IOException;
+import java.util.List;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.disq_bio.disq.impl.formats.vcf.AbstractVcfSink;
+import org.disq_bio.disq.impl.formats.vcf.VcfFormat;
+import org.disq_bio.disq.impl.formats.vcf.VcfSink;
+import org.disq_bio.disq.impl.formats.vcf.VcfSinkMultiple;
+import org.disq_bio.disq.impl.formats.vcf.VcfSource;
+
+/** The entry point for reading or writing a {@link HtsjdkVariantsRdd}. */
+public class HtsjdkVariantsRddStorage {
```

Comment:
There is a lot of paralelism in this class with the reads counterpart - maybe an abstract class would keep in sync both (I saw that there is no NIO support in variants yet, but it could throw by now).

Reply:
That makes sense too.

Reply:
Looking again, there's some commonality, but I'm not sure how useful it is to have an abstract base class at this point. Would it be a distraction to the user looking at javadoc if some methods are in a base class?

Reply:
I believe that the javadoc can show the base class methods, and the javadoc can always be overriden even if the implementation is to call the super method.